output/typescript/types.ts

```diff
@@ -10626,10 +10634,11 @@ export interface MlDetector {
   detector_index?: integer
   exclude_frequent?: MlExcludeFrequent
   field_name?: Field
-  function: string
+  function?: string
```

Comment:
This interface needs splitting into 2.  When creating a job for the first time `function` is not compulsory, but when getting the definition back from the get jobs API `function` will be present for every detector.